packages/api/internal/batcher/batcher.go

```diff
@@ -0,0 +1,71 @@
+package batcher
+
+import (
+	"context"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/e2b-dev/infra/packages/shared/pkg/db"
+)
+
+type Batcher struct {
+	db        *db.DB
+	templates map[string]int64
+	ctx       context.Context
+
+	mu sync.Mutex
+}
+
+func NewBatcher(ctx context.Context, db *db.DB) *Batcher {
+	b := &Batcher{
+		db:        db,
+		ctx:       ctx,
+		templates: make(map[string]int64),
+	}
+	go b.Loop()
+
+	return b
+}
+
+// UpdateTemplateSpawnCount updates the spawn count for the given environment.
+func (b *Batcher) UpdateTemplateSpawnCount(templateID string) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	if _, ok := b.templates[templateID]; !ok {
+		b.templates[templateID] = 0
+	}
+
+	b.templates[templateID]++
+}
+
+func (b *Batcher) Loop() {
+	ticker := time.NewTicker(time.Minute)
+	for {
+		select {
```

Comment:
I would add `defer ticker.Stop()` because the loop actually can stop because of the context. Also the `Loop` method could be private, because it is called only in the init.